### PR TITLE
feat: add insights module

### LINKED ISSUE #168: Add insights module
# Feature Request

## Wanted Module
An insights module.

## Endpoint (if you found one)
`https://query2.finance.yahoo.com/ws/insights/v2/finance/insights`

### Query Params
| Name | Type | Required |
| ---- | ---- | -------- |
| symbol     | string     | true          |
| lang | string | false|
| region | string | false |
| reportsCount | number | false |
| getAllResearchReports | boolean | false |

## Potential Example Usage
This module *could* be used in the following way:
```ts
yf.insights("AMD", { reportsCount: 5 });
```

### PR DESCRIPTION
**Pull request by PythonCreator27**
Closes #168.

## Changes
- Add insights module
- Add tests
- Add docs

## Type
- [x] New Module
- [ ] Other New Feature
- [ ] Validation Fix
- [ ] Other Bugfix
- [x] Docs
- [x] Chore/other

## Comments/notes
~~Currently, I cannot run tests locally, see https://github.com/gadicc/node-yahoo-finance2/pull/144#issuecomment-835503904. I am opening the PR early so that I can see CI run.~~

### CODE REVIEW

**Comment by gadicc** [comment]
Hey @PythonCreator27, thanks so much, this is fantastic work as usual!  Thanks for yet another module!

I'm going to hold off merging a bit just until we solve the other issues, so we can have consistency on the `.js` extensions and just in case we end up reverting the esm stuff.

**Comment by PythonCreator27** [comment]
@gadicc Correct me if I am wrong, but now that the whole tests issue is sorted out, I think that this can be merged. The JSON imports should not matter, as any changes would only affect files that import JSON, which this module does not (directly).

**Comment by gadicc** [comment]
Hey @PythonCreator27, you're absolutely right.  Sorry, slipped me mind with my upcoming trip.  Now that we know we're keeping ESM can I just ask that all the imports are on `.js` for consistency.  I'll tag them inline.  Thanks!  And of course, thanks again for contributing another awesome module!

File: src/modules/insights.spec.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import insights from "./insights";
+import { testSymbols } from "../../tests/symbols";
+
+import testYf from "../../tests/testYf";
```
Comment (gadicc) [inline src/modules/insights.spec.ts:4]: here

File: src/other/quoteCombine.spec.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { jest } from "@jest/globals";
 
-import quoteCombine from "./quoteCombine.js";
-import testYf from "../../tests/testYf.js";
+import quoteCombine from "./quoteCombine";
+import testYf from "../../tests/testYf";
```
Comment (gadicc) [inline src/other/quoteCombine.spec.ts:4]: and here

**Review by gadicc — CHANGES_REQUESTED** [review]

**Comment by PythonCreator27** [comment]
Oops. Fixed now. Also generated json `http` files. @gadicc Anything else?

**Comment by gadicc** [comment]
Awesome, @PythonCreator27, thanks!  I think maybe you didn't commit your fix on `quoteCombine.spec.ts` but it's such an easy fix that I'm going to merge and fix my end to avoid wasting any more time because of the timezone difference between us.  One last big final thanks from me for another module!

**Pull request merged**